### PR TITLE
added ipfs.ls test spec

### PR DESCRIPTION
**Pull request by fazo96**
While working with the API I noticed weird behavior in `ipfs.ls`. Sometimes it calls the callback twice and some other times when there's an error the error object is undefined and the response object is the actual error.

After talking briefly with @Dignifiedquire I've been trying to write tests to test these and also normal `ipfs.ls` functionality. Please let me know about anything wrong with the PR :+1:

Looks like I couldn't reproduce the issue I was having. I'll try again if I happen to run into it, but it was probably some mistake on my end.

### CODE REVIEW

File: tasks/daemons.js [hunk 1]
```diff
@@ -70,11 +70,13 @@ function startDisposableDaemons (callback) {
               throw err
             }
 
-            ipfsNodes[key].startDaemon((err, ignore) => {
+            ipfsNodes[key].startDaemon((err, daemon) => {
               if (err) {
                 throw err
               }
-
+              daemon.add('test/test-folder', {r: true}, (e, r) => {
```
Comment (diasdavid) [inline tasks/daemons.js]: This folder is added too during the `add` tests, could you use the hash from there?

**Comment by diasdavid** [comment]
Thank you @fazo96 :)

I had in my memory that we had `ls` tests, in fact, it was always our sluggish test, but I see that with the refactoring, it was lost (//cc @Dignifiedquire ). You can find the old test here: https://github.com/ipfs/js-ipfs-api/blob/bdf2e38ec233707d16c6fbd55ce878cb91be1987/test/tests.js#L178-L230 . Your tests are more complete, it could only also confirm that all of the links returned are the ones expected, would you mind adding that?

Thank you!

File: test/api/ls.spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    apiClients['a'].ls('QmTDH2RXGn8XyDAo9YyfbZAUXwL1FCr44YJCN9HBZmL9Gj', (err, res) => {
+      if (err === null && res !== undefined) {
+        assert(res.Objects[0].Links[0])
+        done()
+      }
+    })
+  })
+  it('should correctly handle a nonexisting hash', function (done) {
+    apiClients['a'].ls('surelynotavalidhashheh?', (err, res) => {
+      if (err !== null && res === undefined) {
```
Comment (Dignifiedquire) [inline test/api/ls.spec.js]: We since switched to `expect`, so you can do something like 

```js
expect(err).to.exist
expect(res).to.not.exist
```

**Comment by Dignifiedquire** [comment]
Upps sorry for removing the `ls` tests :(

**Comment by fazo96** [comment]
Will fix it soon :+1: Thanks!

**Comment by diasdavid** [comment]
@fazo96 still update to do the last changes so that we can merge this PR?  Thank you :)

**Pull request closed without merging**